src/test/java/com/theelixrlabs/healthcare/service/PatientServiceTest.java

```diff
@@ -0,0 +1,73 @@
+package com.theelixrlabs.healthcare.service;
+
+import com.theelixrlabs.healthcare.constants.PatientConstants;
+import com.theelixrlabs.healthcare.exceptionHandler.PatientException;
+import com.theelixrlabs.healthcare.repository.PatientRepository;
+import com.theelixrlabs.healthcare.utility.MessageUtil;
+import com.theelixrlabs.healthcare.validation.Validator;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import java.util.UUID;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.times;
+
+public class PatientServiceTest {
+
+    @Mock
+    private PatientRepository patientRepository;
+
+    @Mock
+    private MessageUtil messageUtil;
+
+    @Mock
+    private Validator validator;
+
+    @InjectMocks
+    private PatientService patientService;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void testDeletePatientById_Success() throws Exception {
+        String patientId = "641cb90a-9d64-468d-8e17-7efdd26482c3";
+        UUID validPatientId = UUID.fromString(patientId);
+        when(validator.validateAndConvertToUUID(anyString(), anyString())).thenReturn(validPatientId);
+        when(patientRepository.existsById(validPatientId)).thenReturn(true);
+        PatientService spyPatientService = spy(patientService);
+        doReturn(false).when(spyPatientService).isPatientAssignedToDoctor(validPatientId);
+        doNothing().when(patientRepository).deleteById(validPatientId);
+        when(messageUtil.getMessage(anyString(), any(Object[].class))).thenReturn("Patient deleted successfully");
+        String testResult = spyPatientService.deletePatientById(patientId);
+        assertEquals("Patient deleted successfully", testResult);
```

Comment:
"Patient deleted successfully"
can be kept inside a variable and used as xpectedResult

Reply:
Resolved.